File: front/src/models/entities/jobOffer.ts
/**
 * A job offer.
 */
export type JobOffer = {
  /**
   * The id of the job offer.
   */
  id: string;

  /**
   * The title of the job offer.
   */
  title: string;

  /**
   * The description of the job offer.
   */
  description: string;

  /**
   * The start date of the job offer.
   */
  startDate: string;

  /**
   * The end date of the job offer.
   */
  endDate: string;

  /**
   * The geographic area of the job offer.
   */
  geographicArea: string;

  /**
   * The geographic area of the job offer.
   */
  status: string;
};

File: backend/api_gateway/src/main/kotlin/com/linkedout/backend/model/JobOffer.kt
package com.linkedout.backend.model

import java.time.LocalDate
import java.util.*

data class JobOffer(
    val id: String,
    val title: String,
    val description: String,
    val startDate: LocalDate,
    val endDate: LocalDate,
    val geographicArea: String,
    val job: Job,
    val company: Company,
    val salary: Int,
    val status: Int,
    val employerId: String
)
